### DISCUSSION TITLE
How does Ironwail get such nice dark environmental shadows?

### DISCUSSION THREAD

**Discussion by CorpusBava** [andrei-drexler/ironwail#257] [Q&A]
Recently, I was forced to use QSS because a certain community map pack (Peril) would go wacky with the clipping if there were long elevator or train rides using Ironwail (damage to player).  Otherwise, IW is my favorite Quake Source port by far and I use it for everything else.  However, I noticed QSS looks crappy compared to Ironwail even when both use_texturemode "GL_NEAREST_MIPMAP_LINEAR" for that nice raw pixel look.  My question is: how does IW get those nice black shadows in environments?  On QSS, dark areas look murky with slight banding effect rather than the desired inky black shadows of IW.  Is there a setting that I can use to make QSS look more like Ironwail?

**Comment by andrei-drexler** [comment]
IW uses a 10-bit framebuffer for intermediate rendering, plus a tiny (almost imperceptible) amount of dithering to avoid banding, which makes fog in dark areas look as smooth as possible. AFAIK there's no equivalent for this in QSS.

Related: https://github.com/sezero/quakespasm/issues/76
